bokehjs/src/coffee/models/glyphs/glyph.ts

```diff
@@ -236,10 +236,33 @@ export abstract class GlyphView extends View {
         xname = `_${xname}`
         yname = `_${yname}`
 
-        if (xr instanceof FactorRange)
-          self[xname] = xr.v_synthetic(self[xname])
-        if (yr instanceof FactorRange)
-          self[yname] = yr.v_synthetic(self[yname])
+        // TODO (bev) more robust detection of multi-glyph case
+        // hand multi glyph case
+        if (self._xs != null) {
+          if (xr instanceof FactorRange) {
+            const result = []
+            for (const arr of self[xname])
+              result.push(xr.v_synthetic(arr))
+            self[xname] = result
+          }
+          if (yr instanceof FactorRange) {
+            const result = []
+            for (const arr of self[yname])
+              result.push(yr.v_synthetic(arr))
+            self[yname] = result
```

Comment:
Simpler and faster:

`import {map} from "core/util/arrayable"`

`self[yname] = map(self[yname], (arr) => yr.v_synthetic(arr))`

Reply:
Thanks I will change

Reply:
get an error:

> Argument of type '{}' is not assignable to parameter of type 'Arrayable<string | number | [string, string] | [string, string, string] | [string, number] | [string, string, number] | [string, string, string, number]>'. Property 'length' is missing in type '{}'.'

Reply:
This works:
```
if (self._xs != null) {
          if (xr instanceof FactorRange) {
            self[xname] = map(self[xname], (arr: any) => xr.v_synthetic(arr))
          }
          if (yr instanceof FactorRange) {
            self[yname] = map(self[yname], (arr: any) => yr.v_synthetic(arr))
          }
        }
```

Reply:
Right, this is the effect of allowing `self: any`. I think, for now, `map(self[yname] as any, ...)` should help overcome this error.

Reply:
Your version is also fine.